comps/agent/langchain/src/agent.py

```diff
@@ -0,0 +1,300 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import List, Tuple, TypedDict, Union
+
+from langgraph.graph import END, StateGraph
+
+from .tools import get_tools_descriptions
+from .utils import has_multi_tool_inputs, setup_llm, tool_renderer
+
+
+def instantiate_agent(args, strategy="react"):
+    if strategy == "react":
+        return ReActAgentwithLangchain(args)
+    elif strategy == "planexec":
+        return PlanExecuteAgentWithLangGraph(args)
+    else:
+        return BaseAgent(args)
+
+
+class BaseAgent:
+    def __init__(self, args) -> None:
+        self.llm_endpoint = setup_llm(args)
+        self.tools_descriptions = get_tools_descriptions(args.tools)
+        self.app = None
+        print(self.tools_descriptions)
+
+    def compile(self):
+        pass
+
+    def execute(self, state: dict):
+        pass
+
+
+class BaseAgentState(TypedDict):
+    input: str
+    date: str
+    plan: Union[List[dict], str]
+    plan_errors: List[str]
+    past_steps: List[Tuple]  # Annotated[List[Tuple], operator.add]
+    # num_replan: int
+    # num_rewrite: int
+    # new_plan: List[dict]
+    response: str
+
+
+class ReActAgentwithLangchain(BaseAgent):
+    def __init__(self, args):
+        super().__init__(args)
+        from langchain import hub
+        from langchain.agents import AgentExecutor, create_react_agent
+
+        prompt = hub.pull("hwchase17/react")
```

Comment:
maybe we should maintain this prompt in this comps instead of directly import from langchain